docs/current_docs/manuals/developer/cookbook.mdx

```diff
@@ -0,0 +1,172 @@
+---
+slug: /manuals/developer/cookbook
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Cookbook
+
+## Filesystem
+
+### Add a local directory or remote repository to a container
+
+The following Dagger Function accepts a `Directory` argument, which could reference either a directory from the local filesystem or from a remote Git repository. It writes the specified directory to a container at the `/src` container path and returns the modified container.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./cookbook/snippets/write-dir/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./cookbook/snippets/write-dir/python/main.py
+```
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```typescript file=./cookbook/snippets/write-dir/typescript/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+Example usage:
```

Comment:
This is a change from the previous cookbook. WDYT of it?

Reply:
I don't think it's necessary, but I'm not strongly against it either. It may not make sense for certain examples (which is fine) and may lead us to trying to show cli-oriented examples in cookbook (which is avoidable)

Reply:
Your thoughts @helderco @levlaz @shykes ?

Reply:
I kind of like it, but I think we should think about adding two parts here. 

1. Example SDK usage (maybe even show how to use in other SDKs) 
2. Example CLI usage (or make it clear that is what is happening) 

I think this is especially important because there are some things that are different between CLI and SDK. For example, in go, optional args are not actually optional if you are using them via the SDK. 

Risk here is that it is possible this becomes too verbose.

Reply:
AFAIK we are de-emphasizing the SDK use case, so IMHO we should not show that part for the moment.

Reply:
I vote ship it and we can start the migration and make a decision as we go